retrieval/discovery/ec2.go

```diff
@@ -110,12 +118,34 @@ func (ed *EC2Discovery) refresh() (*config.TargetGroup, error) {
 				labels := model.LabelSet{
 					ec2LabelInstanceID: model.LabelValue(*inst.InstanceId),
 				}
-				if inst.PublicIpAddress != nil {
-					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
-				}
 				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
 				addr := fmt.Sprintf("%s:%d", *inst.PrivateIpAddress, ed.port)
 				labels[model.AddressLabel] = model.LabelValue(addr)
+
+				if inst.PublicIpAddress != nil {
+					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
+					labels[ec2LabelPublicDNS] = model.LabelValue(*inst.PublicDnsName)
+				}
+
+				labels[ec2LabelAZ] = model.LabelValue(*inst.Placement.AvailabilityZone)
+
+				if inst.VpcId != nil {
+					labels[ec2LabelVPCID] = model.LabelValue(*inst.VpcId)
+
+					subnetsMap := make(map[string]struct{})
+					for _, eni := range inst.NetworkInterfaces {
+						subnetsMap[*eni.SubnetId] = struct{}{}
+					}
+					subnets := []string{}
+					for k := range subnetsMap {
+						subnets = append(subnets, k)
+					}
+					labels[ec2LabelSubnetID] = model.LabelValue(
+						ed.subnetSeparator +
```

Comment:
This looks wrong: why is the comma both before and after the subnets list? Also, why not assign to multiple labels here instead of joining a bunch? See the tags for an example. (Note: I haven't actually seen what the subnets are, so I'm guessing about this, and could be way wrong.)